### PR TITLE
Pickled flag reorg; SI-5504

### PR DESCRIPTION
**Pull request by odersky**
Cleaned up and optimized code that maps between raw and pickled flags. Avoids mystery constants. Makes a whole bunch of new flags be pickled which were not pickled before (more precisely: Everything in InitialFlags with value greater than 1 << 31 which is not in FlagsNotPickled now gets pickled whereas before it wasn't. Among these: VARARGS, IMPLCLASS, SPECIALZED, DEFAULTINIT, SYNCHRONIZED. I am curious how many tickets will get fixed by this change.

The first one I noted is t5504, which previously enforced the buggy behavior through a neg check!

There are also some build manager check file changes that have to do with the fact that flags now print in a different order for performance reasons.

review by @gregk or @paulp

### CODE REVIEW

**Comment by paulp** [comment]
The test for t5504 is there because I went to great lengths to improve the error message from an unpositioned "error: type _$1 is defined twice" to something which pointed at the problem.  There is no other way to have tests keeping an eye on such changes than to put them in neg; it doesn't imply the behavior is correct, only that if the behavior changes, someone should be made aware of the change.

File: src/reflect/scala/reflect/internal/Flags.scala [hunk 1]
```diff
@@ -279,6 +280,12 @@ class Flags extends ModifierFlags {
   /** Module flags inherited by their module-class */
   final val ModuleToClassFlags = AccessFlags | TopLevelCreationFlags | CASE | SYNTHETIC
 
+  /** These flags are not pickled */
+  final val FlagsNotPickled = IS_ERROR | OVERLOADED | LIFTED | TRANS_FLAG | LOCKED | TRIEDCOOKING
```
Comment (paulp) [inline src/reflect/scala/reflect/internal/Flags.scala:284]: How about adding something like

```
// A precaution against future additions to FlagsNotPickled turning out
// to be overloaded flags thus not-pickling more than intended.
assert((OverloadedFlagsMask & FlagsNotPickled) == 0, flagsToString(OverloadedFlagsMask & FlagsNotPickled))
```

**Comment by odersky** [comment]
I agree the change would be good. Can somebody please get this unstuck? It merges with my master, but it seems not to merge with 2.10.x. I can't figure out even how to get a local 2.10.x :-( This is really annoying. Git is harder than I anticipated. But the patch is hugely important and given the time people have already wasted on this should go in ASAP. Thanks!

**Comment by paulp** [comment]
I settled things with 2.10.x and reissued it as https://github.com/scala/scala/pull/887 .

**Comment by adriaanm** [comment]
@odersky, in the following, replace origin (twice) by the name you gave to the github.com/scala/scala remote

```
git fetch origin ; git checkout -t origin/2.10.x
```

I can stop by your office later for more details if you like.

**Pull request closed without merging**